llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -22809,7 +22809,8 @@ static SDValue scalarizeExtractedBinOp(SDNode *ExtElt, SelectionDAG &DAG,
 
   EVT ResVT = ExtElt->getValueType(0);
   if (Opc == ISD::SETCC &&
-      (ResVT != Vec.getValueType().getVectorElementType() || LegalTypes))
+      (ResVT != Vec.getValueType().getVectorElementType() || ResVT != MVT::i1 ||
```

Comment:
Just check TLI.getBooleanContents?

Reply:
Rather than bail out, I've just added support for detecting the additional sign-extend required.